homeassistant/components/deako/config_flow.py

```diff
@@ -0,0 +1,126 @@
+"""Config flow for deako."""
+
+import logging
+from typing import Any
+
+from pydeako.deako import Deako, DeviceListTimeout, FindDevicesTimeout
+from pydeako.discover import DeakoDiscoverer, DevicesNotFoundException
+import voluptuous as vol
+
+from homeassistant.components import zeroconf
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST
+from homeassistant.core import callback
+
+from .const import ADDRESS, CONNECTION, DEFAULT_PORT, DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class DeakoConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a Deako integration config flow."""
+
+    VERSION = 1
+    connection: Deako | None = None
+
+    address: str | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self._async_show_setup_form()
+
+        await self.async_set_unique_id(DOMAIN)
+        host = user_input.get(CONF_HOST)
+        if host is not None:
+            self.address = f"{host}:{DEFAULT_PORT}"
+            self._abort_if_unique_id_configured(
+                updates={ADDRESS: self.address},
+                reload_on_update=True,
+            )
+        else:
+            self._abort_if_unique_id_configured()
+
+        return await self._finalize()
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle zeroconf discovery."""
+        # only one instance of deako integration should exist, device is a bridge
+        if self.hass.data.get(DOMAIN) is not None:
+            return self.async_abort(reason="already_configured")
+
+        await self.async_set_unique_id(DOMAIN)
```

Comment:
Why do we enforce this? Why don't we allow if someone uses 2 deako bridges?

Reply:
Good question. We only need one device for a successful integration. A second bridge won't be beneficial or net any other devices. 

I tested removing this constraint and on larger installations, the discovered devices menu is flooded with all the devices that could be the bridge.

There's one instance I can see for two bridges and that would be separate installations on the same network. So you would have two sets of devices, but there's no way to distinguish what devices are a part of what installation from the advertisement data and even with the local API, there's no way to determine if a device is part of installation A or installation B without comparing the device list of multiple devices.

Reply:
Can you elaborate on this? 
> the discovered devices menu is flooded with all the devices that could be the bridge.

Reply:
Yeah so I have access to a larger installation of Deako devices (>100) and due to their behavior with getting online to check in, there are always more than 10 online at a time roughly. 

By removing the constraint that there can only be one active config flow via the DOMAIN, the discovered section in the integrations menu will now have all the devices listed as separate Deako integrations. We only need one to act as the bridge, so having all those listed there is misleading.

Reply:
And why should we connect to a bridge if we can also talk to the device directly?

Reply:
Not every device is guaranteed to be on the network and often most devices are not. The bridge device gives us two things:

1. the list of all known devices, whether on the network or not
2. access to the encrypted BLE mesh network for control requests

Reply:
How can we detect if the device is a bridge?

Reply:
Good question! Any device that's on the network _can_ be the bridge device. The device that _is_ selected as the bridge device (aka connected to with a socket) will no longer advertise I believe.  I would have to confirm with the firmware team.

Reply:
No but how do you know if the zeroconf message is the one sent by the bridge?

Reply:
All of the zeroconf advertisements are sent from devices that aren't the bridge, just ones that could be the bridge. This flow is essentially bridge selection where the first device that advertises is the bridge.

Reply:
Mind sharing the zeroconf debug logs? Like I think I start to know what you're getting at, but I want to be sure.

Reply:
So I talked to firmware team, device continues to advertise when it's connected to as the bridge, but I think that's fine for the integration's use cases.

I'll get you some zeroconf logs on the larger installation that I have access to.

Reply:
Here are the logs:

[2024-07-17 11:06:18.log](https://github.com/user-attachments/files/16272949/2024-07-17.11.06.18.log)

Reply:
Can you do `homeassistant.components.zeroconf: debug` instead?

Reply:
You bet! Much easier to read. 

[2024-07-17 12:51:08.log](https://github.com/user-attachments/files/16274057/2024-07-17.12.51.08.log)